PROTO_tests/tests/client_dtypes_test.py

```diff
@@ -0,0 +1,178 @@
+        assert isinstance(bv, client_dtypes.BitVector)
+        assert bv.to_list() == bv_rev_answer
+        assert bv.dtype == ak.bitType
+
+        # test use of vectorizer function
+        arr = ak.arange(4, dtype=int_types)
+        bvectorizer = ak.BitVectorizer(3)
+        bv = bvectorizer(arr)
+        assert isinstance(bv, client_dtypes.BitVector)
+        assert bv.to_list() == bv_answer
+        assert bv.dtype == ak.bitType
+
+    def test_bit_vector_error_handling(self):
+        # fail on argument types
+        with pytest.raises(TypeError):
+            ak.BitVector(17, width=4)
+        arr = ak.array([1.1, 8.3])
+        with pytest.raises(TypeError):
+            ak.BitVector(arr)
+
+    @pytest.mark.parametrize("int_types", INT_TYPES)
+    def test_bit_vector_edge_case(self, int_types):
+        bv_answer = [
+            "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||..",
+            "||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.|",
+            "|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||.",
+        ]
+
+        arr = ak.arange(2**64 - 4, 2**64 - 1, dtype=int_types)
+        bv = ak.BitVector(arr, width=64)
+        assert isinstance(bv, client_dtypes.BitVector)
+        assert bv.to_list() == bv_answer
+        assert bv.dtype == ak.bitType
+
+    def test_field_creation(self):
+        values = ak.arange(4)
+        names = ["8", "4", "2", "1"]
+        f = ak.Fields(values, names)
+        assert isinstance(f, ak.Fields)
+        assert f.to_list() == ["---- (0)", "---1 (1)", "--2- (2)", "--21 (3)"]
+        assert f.dtype == ak.bitType
+
+        # Named fields with reversed bit order
+        values = ak.array([0, 1, 5, 8, 12])
+        names = ["Bit1", "Bit2", "Bit3", "Bit4"]
+        f = ak.Fields(values, names, MSB_left=False, separator="//")
+        expected = [
+            "----//----//----//----// (0)",
+            "Bit1//----//----//----// (1)",
+            "Bit1//----//Bit3//----// (5)",
+            "----//----//----//Bit4// (8)",
+            "----//----//Bit3//Bit4// (12)",
+        ]
+        assert f.to_list() == expected
+
+        values = ak.arange(8, dtype=ak.uint64)
+        names = [f"Bit{x}" for x in range(65)]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", "t"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", ""]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["abc", "123"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, separator="abc")
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, pad="|!~", separator="//")
+
+    @pytest.mark.parametrize("int_types", INT_TYPES)
+    def test_ipv4_creation(self, int_types):
```

Comment:
It would be beneficial to have a "general" version of this where we also use `pytest.prob_size` and create IPs over some range, or even randomly generated integers. While this test is good for testing known examples, we need to expand the coverage a bit. I would recommend either adding an additional test or replacing this one in order to get better coverage.

Reply:
Updated